// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package video_tutorials.proto;

// Represents the feature where the video tutorial targeted to.
enum FeatureType {
  INVALID = 0;
  DEBUG = 1;
  DOWNLOAD = 2;
  SEARCH = 3;
  // For test only.
  TEST = 9999999;
}

// Metadata about a language.
message Language {
  // The locale associated with the lnaguage.
  string locale = 1;

  // Name of the language.
  string name = 2;

  // Name of the language in native text.
  string native_name = 3;
}

// A group of video tutorials with same language.
message VideoTutorialGroup {
  // Language of the video.
  Language language = 1;

  // A list of video tutorials metadata.
  repeated VideoTutorial tutorials = 2;
}

// Metadata of a video tutorial.
message VideoTutorial {
  // Type of feature where this video tutorial targeted.
  FeatureType feature = 1;

  // The user visible description of this video tutorial.
  string title = 2;

  // The complete URL for the video.
  string video_url = 3;

  // Youtube Video URL for sharing.
  string share_url = 4;

  // The complete URL of the thumbnail image.
  string poster_url = 5;

  // The URL of this video's caption.
  string caption_url = 6;

  // Duration of the video in seconds.
  int64 video_length = 7;
}

// Response from the server containing tutorial groups for all languages.
message ServerResponse {
  // Tutorials for all available languages.
  repeated VideoTutorialGroup tutorial_groups = 1;
}